modules/git/blame.go

```diff
@@ -99,51 +95,39 @@ func (r *BlameReader) NextPart() (*BlamePart, error) {
 
 // Close BlameReader - don't run NextPart after invoking that
 func (r *BlameReader) Close() error {
-	defer r.finished() // Only remove the process from the process table when the underlying command is closed
-	r.cancel()         // However, first cancel our own context early
-
+	err := <-r.done
+	_ = r.reader.Close()
 	_ = r.output.Close()
-
-	if err := r.cmd.Wait(); err != nil {
-		return fmt.Errorf("Wait: %w", err)
-	}
-
-	return nil
+	return err
 }
 
 // CreateBlameReader creates reader for given repository, commit and file
 func CreateBlameReader(ctx context.Context, repoPath, commitID, file string) (*BlameReader, error) {
-	return createBlameReader(ctx, repoPath, GitExecutable, "blame", commitID, "--porcelain", "--", file)
-}
-
-func createBlameReader(ctx context.Context, dir string, command ...string) (*BlameReader, error) {
-	// Here we use the provided context - this should be tied to the request performing the blame so that it does not hang around.
-	ctx, cancel, finished := process.GetManager().AddContext(ctx, fmt.Sprintf("GetBlame [repo_path: %s]", dir))
-
-	cmd := exec.CommandContext(ctx, command[0], command[1:]...)
-	cmd.Dir = dir
-	cmd.Stderr = os.Stderr
-	process.SetSysProcAttribute(cmd)
-
-	stdout, err := cmd.StdoutPipe()
+	cmd := NewCommandContextNoGlobals(ctx, "blame", CmdArgCheck(commitID), "--porcelain")
```

Comment:
I think 
```suggestion
	cmd := NewCommandContextNoGlobals(ctx, "blame", AddDynamicArguments(commitID), "--porcelain")
```
is a better fit, as we then don't panic on an invalid commit, and instead handle the error more gracefully.

Reply:
done.